File: Bisection-Method/Bisection.java
import java.util.Scanner;

class Bisection {
    static int a3, a2, a1, a0;

    static float f(float x) {
        return a3 * x * x * x + a2 * x * x + a1 * x + a0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        float xl, xm, xu, fxm, fxl, fxu, E, Era;
        System.out.println("Enter the value of a3,a2,a1 and a0");
        a3 = sc.nextInt();
        a2 = sc.nextInt();
        a1 = sc.nextInt();
        a0 = sc.nextInt();
        System.out.println("Enter the value of xl,xu and Error");
        xl = sc.nextFloat();
        xu = sc.nextFloat();
        E = sc.nextFloat();

        while (true) {
            xm = (xu + xl) / 2;
            fxm = f(xm);
            fxl = f(xl);
            fxu = f(xu);
            if ((fxl * fxm) == 0) {
                System.out.println("Root is: " + xm);
                break;
            } else if ((fxl * fxm) < 0) {
                xu = xm;
            } else {
                xl = xm;
            }
            Era = (xu - xl) / xu;
            if (Math.abs(Era) < E) {
                System.out.println("Root is :" + xm);
                break;
            }

        }

    }
}

File: Bisection-Method/Bisection.py

def f(x,a3,a2,a1,a0):

    return (a3*(x**3))+(a2*(x**2))+(a1*x)+a0

def main():
    a3=int(input("Enter the vale 0f a3 :"))
    a2=int(input("Enter the value of a2 :"))
    a1=int(input("Enter the value of a1 :"))
    a0=int(input("Enter the value of a0 :"))
    xu=int(input("Enter the value of xu :"))
    xl=int(input("Enter the value of xl :"))
    E=float(input("Enter the value of E :"))
    while(True):
        xm=(xu+xl)/2
        fxl=f(xl,a3,a2,a1,a0)
        fxu=f(xu,a3,a2,a1,a0)
        fxm=f(xm,a3,a2,a1,a0)
        if(fxl*fxm==0):
            print("Root is :",xm)
            break
        elif(fxl*fxm<0):
            xu=xm
        else:
            xl=xm   
            
        Era=(xu-xl)/xu
        if(abs(Era)<E):
            print("Root is :",xm)
            break
    
main()        
             
        

        

    
    